src/operator/numpy/np_dot.cc

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_dot.cc
+ * \brief CPU Implementation of numpy-compatible dot
+ */
+
+#include "./np_dot-inl.h"
+
+namespace mxnet {
+namespace op {
+
+NNVM_REGISTER_OP(_numpy_dot)
+.describe(R"doc(Dot product of two arrays. Specifically,
+
+- If both a and b are 1-D arrays, it is inner product of vectors (without complex conjugation).
+
+- If both a and b are 2-D arrays, it is matrix multiplication, but using matmul or a @ b is preferred.
```

Comment:
I though this is copied from numpy document. Is `matmul` or `@` operator available in MXNet?

Reply:
There's no `matmul` in mxnet, also I've got no knowledge about `@` operator, can you provide a link?

Reply:
I don't know mxnet has these two operators. So this document is a little misleading to me~

Reply:
Please modify the doc to keep aligned with mxnet functionality.

Reply:
Done.